content/learn/beginner/inputs-1.md

```diff
@@ -0,0 +1,169 @@
++++
+title = "Inputs 1"
+weight = 240
++++
+
+When we were starting to write Custom Commands, we stuck to a somewhat static custom command --- one that doesn't take
+any arguments and has a fixed output. This is a good starting point, but it's not very useful in practice. In this
+chapter, we'll learn how to create custom commands that take inputs.
+
+## Parsing Arguments
+
+Because parsing arguments is no easy task, we provide a convenience function, `parseArgs`, to parse the arguments passed
+to a custom command. For simple commands, this is the recommended way. In a later chapter, we will explore a more
+hands-on approach to parsing arguments.
+
+### Defining Arguments
+
+The first step is to define the arguments that the command will take. This is done using the aforementioned `parseArgs`
+function. The syntax is as follows:
+
+```go
+{{ $args := parseArgs <"num required args"> <"error message"> ...carg }}
+```
+
+The first argument `num required args` is the number of required arguments. You can use this to tell `parseArgs` how
+many arguments of the given definitions at the end are required. If the number of defined arguments is greater than the
+number of required arguments, the remaining arguments are considered optional. You cannot define optional positional
+arguments using `parseArgs`.
```

Comment:
I was a little confused reading this paragraph as a whole. Why not leave it at the first sentence (which seems perfectly clear to me -- perhaps a short example to drive home the point at most?) The rest I could not quite grok: "how many arguments of the given definitions at the end are required" seems redundant; the "optional" part follows from `!required`, and the "cannot define optional positional arguments" I cannot make sense of.

Reply:
Reading this section back I do agree it is confusingly written and
unnecessarily verbose. I'm not quite sure how to properly introduce the
concept of optional arguments when `num defined args > num required`,
but I'm positive the example further down should illustrate that to a
satisfactory extend.